examples/bert/bert_classifier_main_V2.py

```diff
@@ -0,0 +1,257 @@
+# Copyright 2019 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Example of building a sentence classifier based on pre-trained BERT
+model.
+"""
+
+import argparse
+import functools
+import importlib
+import logging
+import os
+
+import torch
+import torch.nn.functional as F
+
+import texar as tx
+from texar.custom.optimizers import BertAdam
+from texar.modules import BertClassifierV2
+from utils import model_utils
+
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument(
+    "--config_downstream", default="config_classifier",
+    help="Configuration of the downstream part of the model")
+parser.add_argument(
+    "--config_data", default="config_data", help="The dataset config.")
+parser.add_argument(
+    "--output_dir", default="output/",
+    help="The output directory where the model checkpoints will be written.")
+parser.add_argument(
+    "--checkpoint", type=str, default=None,
+    help="Path to a model checkpoint (including bert modules) to restore from.")
+parser.add_argument(
+    "--do_train", action="store_true", help="Whether to run training.")
+parser.add_argument(
+    "--do_eval", action="store_true",
+    help="Whether to run eval on the dev set.")
+parser.add_argument(
+    "--do_test", action="store_true",
+    help="Whether to run test on the test set.")
+
+args = parser.parse_args()
+
+config_data = importlib.import_module(args.config_data)
+
+config_downstream = importlib.import_module(args.config_downstream)
+config_downstream = {
+    k: v for k, v in config_downstream.__dict__.items()
+    if not k.startswith('__')}
+
+logging.root.setLevel(logging.INFO)
+
+
+def main():
+    """
+    Builds the model and runs.
+    """
+
+    tx.utils.maybe_create_dir(args.output_dir)
+
+    # Loads data
+    num_train_data = config_data.num_train_data
+
+    # Builds BERT
+    hparams = {
+        'clas_strategy': 'cls_time'
+    }
+    model = BertClassifierV2(hparams=hparams)
+    if torch.cuda.is_available():
+        model = model.cuda()
```

Comment:
The following might be a better style:
https://github.com/asyml/texar-pytorch/blob/fc99e0db92666caa947aba96e4c950db3aeb70a2/examples/transformer/transformer_main.py#L61-L65

This way you don't have to check for `cuda.is_available()` everywhere, you can just call `.to(device)`. Also, you can pass `device` to `RecordData` constructor (or call `.to` after construction) to automatically move input to the appropriate device.